pkg/datastore/mysql/ensurer/indexes.sql

```diff
@@ -86,11 +86,27 @@ CREATE INDEX deployment_chain_id_updated_at_desc ON Deployment (DeploymentChainI
 
 -- index on `ProjectId` ASC and `CreatedAt` ASC
 CREATE INDEX event_project_id_created_at_asc ON Event (ProjectId, CreatedAt);
+-- index on `ProjectId` ASC and `CreatedAt` DESC
+CREATE INDEX event_project_id_created_at_desc ON Event (ProjectId, CreatedAt DESC);
+-- index on `ProjectId` ASC and `UpdatedAt` DESC
+CREATE INDEX event_project_id_updated_at_desc ON Event (ProjectId, UpdatedAt DESC);
 
 -- index on `EventKey` ASC, `Name` ASC, `ProjectId` ASC and `CreatedAt` DESC
 ALTER TABLE Event ADD COLUMN EventKey VARCHAR(64) GENERATED ALWAYS AS (data->>"$.event_key") VIRTUAL NOT NULL, ADD COLUMN Name VARCHAR(50) GENERATED ALWAYS AS (data->>"$.name") VIRTUAL NOT NULL;
 CREATE INDEX event_key_name_project_id_created_at_desc ON Event (EventKey, Name, ProjectId, CreatedAt DESC);
 
+-- index on `ProjectId` ASC, `Status` ASC, UpdatedAt DESC
+ALTER TABLE Event ADD COLUMN Status INT GENERATED ALWAYS AS (data->>"$.status") VIRTUAL NOT NULL;
+CREATE INDEX project_id_status_updated_at_desc ON Event (ProjectId, Status, CreatedAt DESC);
+
+-- index on `Name` ASC, `ProjectId` ASC, `UpdatedAt` DESC
+ALTER TABLE Event ADD COLUMN Name VARCHAR(50) GENERATED ALWAYS AS (data->>"$.name") VIRTUAL NOT NULL;
```

Comment:
Not sure it would work, since this `ALTER TABLE` command adds a column with the same name as the existing one (column name `Name`) 🤔

Reply:
For sure. Let me look into it.

Reply:
It seems to work fine.
```
mysql> describe Event;
+-----------+--------------+------+-----+---------+-------------------+
| Field     | Type         | Null | Key | Default | Extra             |
+-----------+--------------+------+-----+---------+-------------------+
| Id        | binary(16)   | NO   | PRI | NULL    |                   |
| Data      | json         | NO   |     | NULL    |                   |
| ProjectId | varchar(50)  | NO   | MUL | NULL    | STORED GENERATED  |
| Extra     | varchar(100) | YES  |     | NULL    | STORED GENERATED  |
| CreatedAt | int          | NO   |     | NULL    | STORED GENERATED  |
| UpdatedAt | int          | NO   |     | NULL    | STORED GENERATED  |
| EventKey  | varchar(64)  | NO   | MUL | NULL    | VIRTUAL GENERATED |
| Name      | varchar(50)  | NO   | MUL | NULL    | VIRTUAL GENERATED |
| Status    | int          | NO   |     | NULL    | VIRTUAL GENERATED |
+-----------+--------------+------+-----+---------+-------------------+
9 rows in set (0.01 sec)

mysql> show index from Event;
+-------+------------+-------------------------------------------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
| Table | Non_unique | Key_name                                  | Seq_in_index | Column_name | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment | Visible | Expression |
+-------+------------+-------------------------------------------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
| Event |          0 | PRIMARY                                   |            1 | Id          | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | event_project_id_created_at_asc           |            1 | ProjectId   | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | event_project_id_created_at_asc           |            2 | CreatedAt   | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | event_project_id_created_at_desc          |            1 | ProjectId   | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | event_project_id_created_at_desc          |            2 | CreatedAt   | D         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | event_project_id_updated_at_desc          |            1 | ProjectId   | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | event_project_id_updated_at_desc          |            2 | UpdatedAt   | D         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | event_key_name_project_id_created_at_desc |            1 | EventKey    | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | event_key_name_project_id_created_at_desc |            2 | Name        | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | event_key_name_project_id_created_at_desc |            3 | ProjectId   | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | event_key_name_project_id_created_at_desc |            4 | CreatedAt   | D         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | project_id_status_updated_at_desc         |            1 | ProjectId   | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | project_id_status_updated_at_desc         |            2 | Status      | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | project_id_status_updated_at_desc         |            3 | CreatedAt   | D         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | name_project_id_status_updated_at_desc    |            1 | Name        | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | name_project_id_status_updated_at_desc    |            2 | ProjectId   | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | name_project_id_status_updated_at_desc    |            3 | Status      | A         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
| Event |          1 | name_project_id_status_updated_at_desc    |            4 | UpdatedAt   | D         |           0 |     NULL |   NULL |      | BTREE      |         |               | YES     | NULL       |
+-------+------------+-------------------------------------------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
```

Reply:
I see, I just forgot that I handled the case `mysqlErrorCodeDuplicateColumnName` already 🙈  Thanks for checking 🙏 
https://github.com/pipe-cd/pipecd/blob/master/pkg/datastore/mysql/ensurer/client.go#L66

Reply:
I've been wondering why it worked but that makes sense!